gcloud/pubsub/client.py

```diff
@@ -43,6 +46,28 @@ class Client(JSONClient):
     """
 
     _connection_class = Connection
+    _publisher_api = _subscriber_api = _iam_policy_api = None
+
+    @property
+    def publisher_api(self):
+        """Helper for publisher-related API calls."""
+        if self._publisher_api is None:
+            self._publisher_api = _PublisherAPI(self.connection)
```

Comment:
That's likely a long discussion, this isn't a great place for it. The short answer is that `start()` enters the context managers for each stub, which fires up some gRPC magic (which consumes memory).

AFAIK, it's the `__enter__` that is expensive, not creating the object. (Using the [factory][1] creates a [dynamic stub][2] and that [helper][3] creates an object with a [simple constructor][4].)

Anyhow, no changes needed by you.

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/18c92fe36f405b104ffce38c2b2386756b5c00e0/gcloud/bigtable/client.py#L61
[2]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/18c92fe36f405b104ffce38c2b2386756b5c00e0/gcloud/bigtable/_generated/bigtable_service_pb2.py#L166
[3]: https://github.com/grpc/grpc/blob/241dea56c8f3688c17c01c96edb497ab89fce582/src/python/grpcio/grpc/beta/implementations.py#L285
[4]: https://github.com/grpc/grpc/blob/241dea56c8f3688c17c01c96edb497ab89fce582/src/python/grpcio/grpc/beta/_stub.py#L43